zipline/gens/tradesimulation.py

```diff
@@ -208,17 +204,13 @@ def _call_before_trading_start(self, dt):
     def on_dt_changed(self, dt):
         if self.algo.datetime != dt:
             self.algo.on_dt_changed(dt)
+            self.algo._portfolio = None
```

Comment:
I think these originally lived in `algo.on_dt_changed` on can be put back there.